hops/featurestore_impl/rest/rest_rpc.py

```diff
@@ -219,3 +219,108 @@ def _put_trainingdataset_create_job(job_conf):
+    """
+    headers = {constants.HTTP_CONFIG.HTTP_CONTENT_TYPE: constants.HTTP_CONFIG.HTTP_APPLICATION_JSON}
+    resource_url = (constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_REST_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_PROJECT_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    os.environ[constants.ENV_VARIABLES.HOPSWORKS_PROJECT_ID_ENV_VAR] +
+                    constants.DELIMITERS.SLASH_DELIMITER + constants.REST_CONFIG.HOPSWORKS_FEATURESTORES_RESOURCE +
+                    constants.DELIMITERS.SLASH_DELIMITER + str(featurestore_id) + constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_FEATUREGROUPS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    str(featuregroup_id) + constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_FEATUREGROUPS_XATTRS_RESOURCE)
+    if name is not None:
+        resource_url += constants.DELIMITERS.SLASH_DELIMITER + name
+
+    response = _http(resource_url, method=constants.HTTP_CONFIG.HTTP_GET, headers=headers)
+    results = {}
+    for item in response["items"]:
+        results[item["name"]] = item["value"]
+    return results
+
+
+def _remove_metadata(featurestore_id, featuregroup_id, name):
+    """
+    Makes a REST call to Hopsworks to delete extended metadata attached to a featuregroup
+
+    Args:
+        :featurestore_id: the id of the featurestore
+        :featuregroup_id: the id of the featuregroup
+        :name: the name of the extended attribute
+
+    Returns:
+        None
+
+    Raises:
+        :RestAPIError: if there was an error in the REST call to Hopsworks
+    """
+    headers = {constants.HTTP_CONFIG.HTTP_CONTENT_TYPE: constants.HTTP_CONFIG.HTTP_APPLICATION_JSON}
+    resource_url = (constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_REST_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_PROJECT_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    os.environ[constants.ENV_VARIABLES.HOPSWORKS_PROJECT_ID_ENV_VAR] +
+                    constants.DELIMITERS.SLASH_DELIMITER + constants.REST_CONFIG.HOPSWORKS_FEATURESTORES_RESOURCE +
+                    constants.DELIMITERS.SLASH_DELIMITER + str(featurestore_id) + constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_FEATUREGROUPS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    str(featuregroup_id) + constants.DELIMITERS.SLASH_DELIMITER +
+                    constants.REST_CONFIG.HOPSWORKS_FEATUREGROUPS_XATTRS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER +
+                    name)
+    response = util.send_request(constants.HTTP_CONFIG.HTTP_DELETE, resource_url, headers=headers)
```

Comment:
Why do you need to call send_request yourself instead of using _http?

Reply:
Delete was failing on json decoding since it doesn't return a json object unless an error happen. didn't want to add more if conditions to that method, but, we can probably have another intermediate method _http_int that returns the response object without decoding it to json

Reply:
Adding a check to _http should be fine. We will run into this again. I believe you can test for it with if response.text and only do the json decoding in that block. The main goal with _http is to avoid code duplication so I would really want to keep avoiding that.

Reply:
ok